heat/core/printing.py

```diff
@@ -0,0 +1,175 @@
+import copy
+import io
+import torch
+
+from typing import Dict
+
+__all__ = ["get_printoptions", "set_printoptions"]
+
+
+# set the default printing width to a 120
+_DEFAULT_LINEWIDTH = 120
+torch.set_printoptions(profile="default", linewidth=_DEFAULT_LINEWIDTH)
+
+
+def get_printoptions() -> Dict:
+    """
+    Returns the currently configured printing options.
+    """
+    return copy.copy(torch._tensor_str.PRINT_OPTS.__dict__)
+
+
+def set_printoptions(
+    precision: int = None,
+    threshold: int = None,
+    edgeitems: int = None,
+    linewidth: int = None,
+    profile: int = None,
+    sci_mode: int = None,
+):
+    """
+    Configures the printing options. List of items shamelessly taken from NumPy and PyTorch (thanks guys!).
+
+    Parameters
+    ----------
+    precision: int
+        Number of digits of precision for floating point output (default=4).
+    threshold: int
+        Total number of array elements which trigger summarization rather than full `repr` string (default=1000).
+    edgeitems: int
+        Number of array items in summary at beginning and end of each dimension (default=3).
+    linewidth: int
+        The number of characters per line for the purpose of inserting line breaks (default = 80).
+    profile: str
+        Sane defaults for pretty printing. Can override with any of the above options. Can be any one of `default`,
+        `short`, `full`.
+    sci_mode: bool
+        Enable (True) or disable (False) scientific notation. If None (default) is specified, the value is automatically
+        inferred by HeAT.
+    """
+    torch.set_printoptions(precision, threshold, edgeitems, linewidth, profile, sci_mode)
+
+    # HeAT profiles will print a bit wider than PyTorch does
+    if profile == "default" and linewidth is None:
+        torch._tensor_str.PRINT_OPTS.linewidth = _DEFAULT_LINEWIDTH
+    elif profile == "short" and linewidth is None:
+        torch._tensor_str.PRINT_OPTS.linewidth = _DEFAULT_LINEWIDTH
+    elif profile == "full" and linewidth is None:
+        torch._tensor_str.PRINT_OPTS.linewidth = _DEFAULT_LINEWIDTH
+
+
+__PREFIX = "DNDarray"
+__INDENT = len(__PREFIX)
+
+
+def __str__(dndarray) -> str:
+    """
+    Computes a printable representation of the passed DNDarray.
+
+    Parameters
+    ----------
+    dndarray: DNDarray
+        The array for which to obtain the corresponding string
+    """
+    tensor_string = _tensor_str(dndarray, __INDENT + 1)
+    if dndarray.comm.rank != 0:
+        return ""
+
+    return "{}({}, dtype=ht.{}, device={}, split={})".format(
+        __PREFIX, tensor_string, dndarray.dtype.__name__, dndarray.device, dndarray.split
+    )
+
+
+def _torch_data(dndarray, summarize: bool) -> torch.Tensor:
```

Comment:
missing parameter Type for dndarray

Reply:
I can only provide these, as soon as we have the aliasing in place. Otherwise I have circular imports